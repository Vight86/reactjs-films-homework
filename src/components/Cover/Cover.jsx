import React from 'react';
import propTypes from 'prop-types';
import styles from './Cover.scss';

const Cover = ({ className, children }) => (
  <figure className={styles[className]}>
    {children}
  </figure>
);

Cover.propTypes = {
  className: propTypes.string,
  children: propTypes.element,
};

Cover.defaultProps = {
  className: 'inCard',
  children: <img src="" alt="" />,
};

export default Cover;
