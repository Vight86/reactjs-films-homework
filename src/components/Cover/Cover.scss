@import '../../utils/variables';

.cover {
  position: relative;
  width: 100%;
  margin: 0;
  background-color: $color-secondary;

  &::before {
    display: block;
    content: '';
  }

  & > img {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: 100%;
    object-fit: cover;
    transform: translate(-50%, -50%);
  }
}

.inCard {
  @extend .cover;

  &::before {
    padding-top: 88%;
  }
}

.inHeroSection {
  @extend .cover;

  &::before {
    padding-top: 46%;
  }

  &::after {
    position: absolute;
    bottom: 0;
    left: 0;
    display: block;
    width: 100%;
    padding-top: 16%;
    background: linear-gradient(to top, $color-dark, transparent);
    content: '';
  }
}
