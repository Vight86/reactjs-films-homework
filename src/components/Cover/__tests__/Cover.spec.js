import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Cover from '../index';

let renderer;
let result;

beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(<Cover className="test"><img src="#" alt="test" /></Cover>);
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('render Button component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });
});
