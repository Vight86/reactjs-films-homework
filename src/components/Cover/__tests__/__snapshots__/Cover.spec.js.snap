// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button component render with props 1`] = `
<figure
  className="class"
>
  <img
    alt="test"
    src="test"
  />
</figure>
`;

exports[`render Button component render without props 1`] = `
<figure
  className="inCard"
>
  <img
    alt=""
    src=""
  />
</figure>
`;
